src/components/WalletSelector/WalletSelector.tsx

```diff
@@ -27,32 +30,35 @@ function WalletSelector() {
     error,
   } = context;
 
-  const headerMessage = active ? 'Log in with wallet' : 'Connect your wallet';
-
   const signer = useMemo(() => {
     return library && account ? library.getSigner(account) : undefined;
   }, [library, account]);
 
+  const { logIn } = useAuthActions();
+  const { hideModal } = useModal();
+
+  useEffect(() => {
+    if (account && isLoadingAuth && !isLoggedIn && signer) {
+      signMessageAndAuthenticate(account, signer).then((jwt) => {
+        logIn(jwt);
+        hideModal();
+      });
+    }
```

Comment:
ohhh i see, so `isLoadingAuth` is set from the previous screen so that this effect can go through? if so i'm wondering if there's a better word / separate state we can use

also nitpick: i prefer `async / await` vs `.then` syntax

Reply:
yeah I think we can use a state that's more specific to the WalletSelector modal.

I removed  the use of `isLoadingAuth` for now because I realized that with the new order of things, I didn't actually need to check the state to trigger the authentication methods. So we would just want some state for disabling the UI.
 
I want it to represent a state where the user has clicked on a Wallet option (like metamask). That way we can disable the whole modal and all the Wallet option buttons while the Metamask prompt is open for example